pkg/kv/kvserver/replica_application_result.go

```diff
@@ -418,6 +419,10 @@ func (r *Replica) handleChangeReplicasResult(
 		log.Infof(ctx, "removing replica due to ChangeReplicasTrigger: %v", chng)
 	}
 
+	if r.raftBasicStatusRLocked().RaftState == raft.StateLeader {
+		r.store.metrics.RangeRaftLeaderRemovals.Inc(1)
```

Comment:
This is noted in the commit message:

>The metric is counted separately per replica, since different replicas can have a different view of the Raft leader.

Specifically, I'm interested in whenever a replica thinks the leader was removed from the range, even though the actual leader may currently be someone else.

The motivation here is to know when various code paths that depend on this get triggered, which happens independently on each replica.

Reply:
On second thought, I think you're right, it only makes sense to count this once. Changed it to count when the leader removes itself. This code is run before the config change is applied to Raft, so it will still see itself as the leader (assuming it still is the leader).